cdap-security/src/main/java/io/cdap/cdap/security/authorization/AuthorizerWrapper.java

```diff
@@ -0,0 +1,215 @@
+    try {
+      return authorizer.listPrivileges(principal).stream().map(
+        p -> new GrantedPermission(p.getAuthorizable(), p.getAction().getPermission())
+      ).collect(Collectors.toSet());
+    } catch (Exception e) {
+      throw AuthEnforceUtil.propagateAccessException(e);
+    }
+  }
+
+  @Override
+  public void grant(Authorizable authorizable, Principal principal, Set<? extends Permission> permissions)
+    throws AccessException {
+    try {
+      authorizer.grant(authorizable, principal, getActionSet(permissions, authorizable.getEntityType()));
+    } catch (Exception e) {
+      throw AuthEnforceUtil.propagateAccessException(e);
+    }
+  }
+
+  @Override
+  public void revoke(Authorizable authorizable, Principal principal, Set<? extends Permission> permissions)
+    throws AccessException {
+    try {
+      authorizer.revoke(authorizable, principal, getActionSet(permissions, authorizable.getEntityType()));
+    } catch (Exception e) {
+      throw AuthEnforceUtil.propagateAccessException(e);
+    }
+  }
+
+  @Override
+  public void revoke(Authorizable authorizable) throws AccessException {
+    try {
+      authorizer.revoke(authorizable);
+    } catch (Exception e) {
+      throw AuthEnforceUtil.propagateAccessException(e);
+    }
+  }
+
+  @Override
+  public void enforce(EntityId entity, Principal principal, Permission permission) throws AccessException {
+    try {
+      authorizer.enforce(entity, principal, getAction(permission, entity.getEntityType()));
+    } catch (Exception e) {
+      throw AuthEnforceUtil.propagateAccessException(e);
+    }
+  }
+
+  @Override
+  public void enforce(EntityId entity, Principal principal, Set<? extends Permission> permissions)
+    throws AccessException {
+    try {
+      authorizer.enforce(entity, principal, getActionSet(permissions, entity.getEntityType()));
+    } catch (Exception e) {
+      throw AuthEnforceUtil.propagateAccessException(e);
+    }
+  }
+
+  @Override
+  public void isVisible(EntityId entityId, Principal principal) throws AccessException {
+    try {
+      authorizer.isVisible(entityId, principal);
+    } catch (Exception e) {
+      throw AuthEnforceUtil.propagateAccessException(e);
+    }
+  }
+
+  @Override
+  public Set<? extends EntityId> isVisible(Set<? extends EntityId> entityIds, Principal principal)
+    throws AccessException {
+    try {
+      return authorizer.isVisible(entityIds, principal);
+    } catch (Exception e) {
+      throw AuthEnforceUtil.propagateAccessException(e);
+    }
+  }
+
+  private Set<Action> getActionSet(Set<? extends Permission> permissions, EntityType entityType) {
+    return permissions.stream().map(p -> getAction(p, entityType))
+      .collect(Collectors.toCollection(LinkedHashSet::new));
+  }
+
+  static Action getAction(Permission permission, EntityType entityType) {
+    if (StandardPermission.GET == permission && entityType == EntityType.DATASET) {
```

Comment:
Another minor nit: I noticed that the SecureKey was also READ, so we may need to add a case for that.